packages/portfolio-contract/src/portfolio.flows.ts

```
@@ -178,6 +244,303 @@ export const provideCosmosAccount = async <C extends 'agoric' | 'noble'>(
+        const evmChain = chainName as AxelarChain;
+
+        const axelar = await orch.getChain('axelar');
+        const gmp = { chain: axelar, fee: move.fee?.value || 0n }; // XXX throw if fee missing?
+        const { lca } = await provideCosmosAccount(orch, 'agoric', kit);
+        const gInfo = await provideEVMAccount(evmChain, gmp, lca, ctx, kit);
+        const accountId: AccountId = `${gInfo.chainId}:${gInfo.remoteAddress}`;
+        const pos = kit.manager.providePosition(
+          way.poolKey,
+          'Compound',
+          accountId,
+        );
+        const { denom } = ctx.gmpFeeInfo;
+        const fee = { denom, value: move.fee ? move.fee.value : 0n };
+        const evmCtx: EVMContext<'compound'> = {
+          addresses: ctx.contracts[evmChain],
+          lca,
+          gmpFee: fee,
+          gmpChain: axelar,
+        };
+
+        if ('src' in way) {
+          todo.push({
+            how: way.how,
+            src: { proxy: gInfo },
+            amount,
+            dest: { pos },
+            apply: () => CompoundProtocol.supply(evmCtx, amount, gInfo),
+            recover: () => assert.fail('last step. cannot recover'),
+          });
+        } else {
+          todo.push({
+            how: way.how,
+            src: { proxy: gInfo },
+            amount,
+            dest: { pos },
+            apply: () => CompoundProtocol.withdraw(evmCtx, amount, gInfo),
+            recover: () => CompoundProtocol.supply(evmCtx, amount, gInfo),
+          });
+        }
+        break;
+      }
+
+      // XXX factor out duplication with Compound?
+      case 'Aave': {
+        // XXX move this check up to wayFromSrcToDesc
+        const chainName = 'src' in way ? way.src : way.dest;
+        assert(keys(AxelarChain).includes(chainName));
+        const evmChain = chainName as AxelarChain;
+
+        const axelar = await orch.getChain('axelar');
+        const gmp = { chain: axelar, fee: move.fee?.value || 0n }; // XXX throw if fee missing?
```

Yes, it should throw if the fee is missing. Ideally, the check should happen earlier in the flow, before committing anything.

---

Well, this is a fee for `provideEVMAccount` - the account might already be created.

---

I did add a test for requiring it for the GMP call to supply to a position.